### ISSUE TITLE
applicationServerKey to Uint8Array

### ISSUE THREAD

**Issue by PRR24** [web-push-libs/web-push#558]
According to MDN https://developer.mozilla.org/en-US/docs/Web/API/PushManager/subscribe
applicationServerKey can be either DOMString or ArrayBuffer.
web-push documentation on the other hand asks to convert VAPID public key to Uint8Array using urlBase64ToUint8Array example function.
I can confirm that no conversion is needed for FF 72.
If this is dependent on browser provider/version, please kindly update the documentation.

**Comment by marco-c** [comment]
PRs welcome!

**Comment by PRR24** [comment]
After looking around, I guess this function originates from https://github.com/GoogleChromeLabs/web-push-codelab and everybody has been copying it since Google published the article.
Sadly this repo is inactive and no common information source has any detailed data.

I would recommend to add an asterisk to the document stating that this function may not be necessary (any more).

**Comment by marco-c** [comment]
No, I think at the beginning that was necessary and so both we and that repo implemented the same.

**Comment by PRR24** [comment]
Was not able to find definite answer, but based on the information on hand I would guess the function became obsolete after fully adopting web push standard with Chrome 52.
So probably the function was needed for Chrome <= 51, Opera for Android, Samsung Internet Browser and possibly more Webkit-based browsers of the era.

**Comment by marco-c** [comment]
Good find! We'd need to test other Chromium-based browsers to see when they picked up the fix.

**Comment by Yagasaki7K** [comment]
That's example doesn't works?

```
function urlBase64ToUint8Array(base64String) {
  const padding = '='.repeat((4 - base64String.length % 4) % 4);
  const base64 = (base64String + padding)
    .replace(/-/g, '+')
    .replace(/_/g, '/');
 
  const rawData = window.atob(base64);
  const outputArray = new Uint8Array(rawData.length);
 
  for (let i = 0; i < rawData.length; ++i) {
    outputArray[i] = rawData.charCodeAt(i);
  }
  return outputArray;
}
 
const vapidPublicKey = '<Your Public Key from generateVAPIDKeys()>';
const convertedVapidKey = urlBase64ToUint8Array(vapidPublicKey);
 
registration.pushManager.subscribe({
  userVisibleOnly: true,
  applicationServerKey: convertedVapidKey
});
```
